skimage/measure/fit.py

```diff
@@ -785,54 +790,64 @@ def ransac(data, model_class, min_samples, residual_threshold,
 
     random_state = check_random_state(random_state)
 
-    if min_samples < 0:
-        raise ValueError("`min_samples` must be greater than zero")
+    # in case data is not pair of input and output, male it like it
+    if not isinstance(data, (tuple, list)):
+        data = (data, )
+    data_lens = list(map(len, data))
+    assert len(set(data_lens)) == 1, 'all data have to have the same number of samples,' \
+                                     ' provided: %r' % [d.shape for d in data]
```

Comment:
This seems like a big change in behaviour now, that is unrelated to the original point of this PR? Additionally, @sciunto made the point recently that we should not use assert statements in validation code, as these are removed by Python when using optimized mode. See #4006 for details.